projects/libfido2/build.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash -eu
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Build libcbor, taken from oss-fuzz/projects/libcbor/build.sh
+# Note SANITIZE=OFF since it gets taken care of by $CFLAGS set by oss-fuzz
+cd ${SRC}/libcbor
+patch -l -p0 < ${SRC}/libfido2/fuzz/README
+mkdir build && cd build
+cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=${WORK} -DSANITIZE=OFF ..
+make -j$(nproc) VERBOSE=1
+make install
+
+# Build OpenSSL, taken from oss-fuzz/projects/openssl/build.sh
+cd ${SRC}/openssl
+CONFIGURE_FLAGS=""
+if [[ ${SANITIZER} = memory ]]
+then
+  CONFIGURE_FLAGS="no-asm"
+fi
+./config --debug no-tests ${CFLAGS} --prefix=${WORK} \
+	 --openssldir=${WORK}/openssl ${CONFIGURE_FLAGS}
+make -j$(nproc) LDCMD="${CXX} ${CXXFLAGS}"
+make install_sw
+
+# Building libfido2 with ${LIB_FUZZING_ENGINE} and chosen sanitizer
+cd ${SRC}/libfido2
+mkdir build && cd build
+cmake -DFUZZ=1 -DFUZZ_LDFLAGS=${LIB_FUZZING_ENGINE} \
+      -DPKG_CONFIG_USE_CMAKE_PREFIX_PATH=1 \
+      -DCMAKE_PREFIX_PATH=${WORK} \
+      -DCMAKE_INSTALL_PREFIX=${WORK} \
+      -DCMAKE_BUILD_TYPE=Debug ..
+make -j$(nproc)
+make install
+
+# Prepare ${OUT} with instrumented libs
+mkdir -p ${OUT}/lib
+for lib in `ls ${WORK}/lib/lib*.so*`; do
+    cp ${lib} ${OUT}/lib;
+done
+
+# Fixup rpath in the fuzzers so they use our libs
+for f in `ls fuzz/fuzz_*`; do
+    cp ${f} ${OUT}/
+    fuzzer=$(basename $f)
+    chrpath -r '$ORIGIN/lib' ${OUT}/${fuzzer}
+done
+
```

Comment:
Is it not possible to do static linking? It's generally significantly easier with statically linked libs.

Reply:
It is possible but would require more changes to libfido2's existing fuzzing targets.
I have tested this PR though, the fuzzers run in oss-fuzz images with address and undefined sanitizers and I get coverage reports.